src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -378,19 +386,86 @@ ExpressionSyntax initializer
 			return assignmentExpressions;
 		}
 
-		private bool IsAnAssignmentTo(
+		private AssignmentInfo? GetAssignmentInfoIfToSymbolOrNull(
 			AssignmentExpressionSyntax assignmentSyntax,
 			ISymbol memberSymbol
 		) {
 			var semanticModel = m_compilation.GetSemanticModel( assignmentSyntax.SyntaxTree );
 
-			var leftSideSymbol = semanticModel.GetSymbolInfo( assignmentSyntax.Left )
-				.Symbol;
+			var lhsExpressions = assignmentSyntax.Left switch {
+				TupleExpressionSyntax tuple => tuple.Arguments.Select( arg => arg.Expression ).ToImmutableArray(),
+				_ => ImmutableArray.Create( assignmentSyntax.Left )
+			};
+
+			for( var i = 0; i < lhsExpressions.Length; ++i ) {
+				var lhs = lhsExpressions[ i ];
 
-			return SymbolEqualityComparer.Default.Equals(
-				memberSymbol,
-				leftSideSymbol
-			);
+				var leftSideSymbol = semanticModel.GetSymbolInfo( lhs )
+					.Symbol;
+
+				if( !SymbolEqualityComparer.Default.Equals(
+					memberSymbol,
+					leftSideSymbol
+				) ) {
+					continue;
+				}
+
+				if( lhsExpressions.Length == 1 ) {
+					return AssignmentInfo.Create(
+						model: semanticModel,
+						isInitializer: false,
+						expression: assignmentSyntax.Right
+					);
+				}
+
+				if( semanticModel.GetTypeInfo( assignmentSyntax.Right ).Type is INamedTypeSymbol assignedType
+					&& assignedType.IsTupleType
+				) {

```

Comment:
Handle Value Tuples.

`GetDeconstructionInfo` does return stuff for them, but they don't have a `Method`, just some nested deconstructioninfos with `Identity` conversions - so need to extract the type info separately anyway.